libs/pq_proto/src/lib.rs

```diff
@@ -295,7 +293,7 @@ impl FeStartupPacket {
         // We shouldn't advance `buf` as probably full message is not there yet,
         // so can't directly use Bytes::get_u32 etc.
         let len = (&buf[0..4]).read_u32::<BigEndian>().unwrap() as usize;
-        if len < 4 || len > MAX_STARTUP_PACKET_LENGTH {
+        if !(4..=MAX_STARTUP_PACKET_LENGTH).contains(&len) {
```

Comment:
> I can't say I like this lint's changes.

Yeah, I just banned that check completely in run_clippy.sh in recent 0d8ced85341. Sorry for not belonging there, I just always run clippy and it started complaining of that.

We can change that, but looks like overwhelming majority also astonished by this suggestion :) @funbringer  was also on our side.

Reply:
Ok..

Why do we still need the run_clippy.sh? I thought it was a needed for the old debian packaged rustc. I think we should just use the config file or source files for the configration. Ah... mac os. Why can't it *check* with `--all-features`?

Reply:
No idea. Seems like @SomeoneToIgnore added that, he might know.

Reply:
We used to have a perf feature that enabled some profiling, but it became obsolete due to async and got removed IIRC.

Current `cargo clippy --all --all-features` works and produces the only warning on this place.

Reply:
Oki pprof makes sense as a reason, I'll then move the contents of run_clippy.sh to clippy config file, make run_clippy.sh the normal `cargo clippy ...` not to break anyones workflow with a disclaimer that one doesn't need to use it.

Reply:
The problem with clippy is that sometimes arbitrary people show up and impose their own standards on the whole community. I don't mean to say that the tool or its default lints are bad, but imo a good deal of its lints are highly subjective.

We shouldn't accept anything we don't like just because it's there.

Reply:
I had misunderstood, it seems `clippy.toml` cannot be used to allow lints just to configure a few until `custom_inner_attributes` is stable.

I'll just revert this `RangeFull::contains` change so we can merge this.

Reply:
Revert in [d149387](https://github.com/neondatabase/neon/pull/3781/commits/d149387923ccf4f7626491cc403715d413fe5af5), resolving.